rskj-core/src/test/java/co/rsk/peg/ReleaseTransactionBuilderTest.java

```diff
@@ -390,9 +392,130 @@ public void buildEmptyWalletTo_utxoProviderException() throws InsufficientMoneyE
+        Coin feeForEachOutput = totalFee.div(outputSize - 1); // minus change output
+
+        Coin firstOutputValue =  testEntry1.getAmount().minus(feeForEachOutput);
+        Coin secondOutputValue =  testEntry2.getAmount().minus(feeForEachOutput);
+        Assert.assertEquals(firstOutputValue, btcTx.getOutput(0).getValue());
+        Assert.assertEquals(secondOutputValue, btcTx.getOutput(1).getValue());
+
+        Coin firstOutputFee = testEntry1.getAmount().minus(btcTx.getOutput(0).getValue());
+        Coin secondOutputFee = testEntry2.getAmount().minus(btcTx.getOutput(1).getValue());
+        Assert.assertEquals(firstOutputFee.add(secondOutputFee), totalFee);
+
+        ReleaseTransactionBuilder.BuildResult builtTx = result.get();
+        Coin inputsValue = builtTx.getSelectedUTXOs().stream().map(UTXO::getValue).reduce(Coin.ZERO, Coin::add);
+        Coin totalPegoutAmount = entries.stream().map(ReleaseRequestQueue.Entry::getAmount).reduce(Coin.ZERO, Coin::add);
+
+        TransactionOutput changeOutput = btcTx.getOutput(outputSize - 1); // last output
+
+        // Last output should be the change output to the Federation
+        Assert.assertEquals(federation.getAddress(), changeOutput.getAddressFromP2SH(networkParameters));
+        Assert.assertEquals(inputsValue.minus(totalPegoutAmount), changeOutput.getValue());
+    }
+
+    @Test
+    public void test_VerifyTXFeeIsSpentEquallyForBatchedPegouts_three_pegouts() {
+        List<UTXO> utxos = Arrays.asList(
+            new UTXO(mockUTXOHash("1"), 0, Coin.COIN, 0, false, federation.getP2SHScript()),
+            new UTXO(mockUTXOHash("2"), 0, Coin.COIN, 0, false, federation.getP2SHScript())
+        );
+
+        Wallet thisWallet = BridgeUtils.getFederationSpendWallet(
+            Context.getOrCreate(networkParameters),
+            federation,
+            utxos,
+            false,
+            mock(BridgeStorageProvider.class)
+        );
+
+        ReleaseTransactionBuilder rtb = new ReleaseTransactionBuilder(
+            networkParameters,
+            thisWallet,
+            federation.address,
+            Coin.SATOSHI.multiply(1000),
+            activations
+        );
+
+        ReleaseRequestQueue.Entry testEntry1 = createTestEntry(123, 3);
+        ReleaseRequestQueue.Entry testEntry2 = createTestEntry(456, 4);
+        ReleaseRequestQueue.Entry testEntry3 = createTestEntry(789, 5);
+        List<ReleaseRequestQueue.Entry> entries = Arrays.asList(testEntry1, testEntry2, testEntry3);
+
+        Optional<ReleaseTransactionBuilder.BuildResult> result = rtb.buildBatchedPegouts(entries);
+
+        Assert.assertTrue(result.isPresent());
+
+        BtcTransaction btcTx = result.get().getBtcTx();
+
+        int outputSize = btcTx.getOutputs().size();
+        Coin totalFee = btcTx.getFee();
+        Coin feeForEachOutput = totalFee.div(outputSize - 1); // minus change output
+
+
+        Coin firstOutputValue =  testEntry1.getAmount().minus(feeForEachOutput);
```

Comment:
```suggestion
        // First Output Pays An Extra Satoshi Because Fee Is Even, And Outputs Is Odd
        Coin firstOutputExpectedValue =  testEntry1.getAmount().minus(feeForEachOutput + 1);
```

Reply:
Resolved